src/main/java/com/google/sps/ColorUtils.java

```diff
@@ -0,0 +1,143 @@
+    colorList.add(new ColorName("Purple", 0x93, 0x70, 0xDB));     // MediumPurple
+    colorList.add(new ColorName("Green", 0x00, 0xFA, 0x9A));      // MediumSpringGreen
+    colorList.add(new ColorName("Turquoise", 0x48, 0xD1, 0xCC));  // MediumTurquoise
+    colorList.add(new ColorName("Navy", 0x00, 0x00, 0x80));
+    colorList.add(new ColorName("Orange", 0xFF, 0xA5, 0x00));
+    colorList.add(new ColorName("Orange", 0xFF, 0x45, 0x00));     // OrangeRed
+    colorList.add(new ColorName("Pink", 0xFF, 0xC0, 0xCB));
+    colorList.add(new ColorName("Purple", 0x80, 0x00, 0x80));
+    colorList.add(new ColorName("Red", 0xFF, 0x00, 0x00));
+    colorList.add(new ColorName("Blue", 0x41, 0x69, 0xE1));       // RoyalBlue
+    colorList.add(new ColorName("Brown", 0x8B, 0x45, 0x13));      // SaddleBrown
+    colorList.add(new ColorName("Silver", 0xC0, 0xC0, 0xC0));
+    colorList.add(new ColorName("Blue", 0x87, 0xCE, 0xEB));       // SkyBlue
+    colorList.add(new ColorName("Turquoise", 0x40, 0xE0, 0xD0));
+    colorList.add(new ColorName("Violet", 0xEE, 0x82, 0xEE));
+    colorList.add(new ColorName("White", 0xFF, 0xFF, 0xFF));
+    colorList.add(new ColorName("Yellow", 0xFF, 0xFF, 0x00));
+    return colorList;
+  }
+  
+  /**
+   * Returns the color name of the closest color (to the query color provided as RGB).
+   * To find the closest color, gets the color list, and then computes the distance between 
+   * the query color and every color in the color list, based on their RGB values.
+   */
+  public String getColorNameFromRGB(int r, int g, int b) throws IllegalArgumentException {
+    if (!(0 < r && r < 255) || !(0 < g && g < 255) || !(0 < b && b < 255)) {
+      throw new IllegalArgumentException("RGB values outside [0, 255] range.");
+    } 
+
+    ArrayList<ColorName> colorList = initColorList();
+    
+    // Get the color name of the color with minimum RGB distance to the query color.
+    ColorName closestColorMatch = null;
+
+    int minRGBDistance = Integer.MAX_VALUE;
+    for (ColorName color : colorList) {
+      int currentRGBDistance = color.computeRGBDistance(r, g, b);
+
+      if (currentRGBDistance < minRGBDistance) {
+        minRGBDistance = currentRGBDistance;
+        closestColorMatch = color;
+      }
+    }
+    
+    if (closestColorMatch != null) {
+      return closestColorMatch.getName();
+    } else {
+      return NO_MATCHED_COLOR_MESSAGE;
+    }
+  }
+  
+  /**
+   * Inner class representing a known color with a color name and an RGB value, and handling the 
+   * computation of the distance to a query color.
+   */
+  public class ColorName {
+    private int r, g, b;
+    private String name;
+    
+    public ColorName(String name, int r, int g, int b) {
+      this.r = r;
+      this.g = g;
+      this.b = b;
+      this.name = name;
+    }
+    
+    /** 
+     * Returns the distance between two RGB color values, based on the three-dimensional distance formula.
```

Comment:
You could use a color model which better reflects human perception. Here's a nice read: https://en.wikipedia.org/wiki/Color_difference

Anyway, I would recommend adding a TODO and not doing it in this PR.

Reply:
Yes, I actually looked at that wikipedia page when deciding how to do compute the distance between two colors, and I chose the one that considers R, G, B "linear dimensions defining the color space" :D I will take a closer look at that page to check out other possibilities as well :)